File: algorithms/strings/remove_even_strings/remove_even_strings.py
def remove_even_strings(arr):
    l = len(arr)-1

    for i in range(l,-1,-1):
        if len(arr[i]) % 2 == 0:
            arr.remove(arr[i])
    return arr


arr = ["John","Jane","James","Jan","Jesse","Jane","Joe","Mary","Joan"]
print remove_even_strings(arr)

File: algorithms/strings/remove_even_strings/remove_even_strings.rb
def remove_even_strings(arr)
  l = arr.length - 1

  l.downto(0) do | i |
    if arr[i].length % 2 == 0
      arr.delete_at(i)
    end
  end
  return arr
end

arr = ["John","Jane","James","Jan","Jesse","Jane","Joe","Mary","Joan"]
print remove_even_strings(arr)
puts
